dace/codegen/targets/intel_fpga.py

```diff
@@ -558,24 +601,19 @@ def generate_host_function_epilogue(sdfg, state, host_stream):
     f.wait();
   }""", sdfg, state_id)
         else:
-            # Launch one-by-one and wait for the cl::Events
+            # While spawning the kernel, indicates the synchronization events (if any)
             host_stream.write(
-                """\
-  std::vector<cl::Event> events;
-  for (auto &k : kernels) {
-    events.emplace_back(k.ExecuteTaskFork());
-  }
-  cl::Event::waitForEvents(events);""", sdfg, state_id)
-        host_stream.write(
-            """\
-  const auto end = std::chrono::high_resolution_clock::now();
-  const double elapsedChrono = 1e-9 * std::chrono::duration_cast<std::chrono::nanoseconds>(end - start).count();
-  std::cout << "Kernel executed in " << elapsedChrono << " seconds.\\n" << std::flush;
-}""", sdfg, sdfg.node_id(state))
-
-    def generate_module(self, sdfg, state, name, subgraph, parameters,
-                        module_stream, host_header_stream, host_body_stream):
-
+                f"""\
+  std::vector<cl::Event> {kernel_name}_events;
+  for (auto &k : {kernel_name}_kernels) {{
+    {kernel_name}_events.emplace_back(k.ExecuteTaskFork({f'{kernel_deps_name}.begin(), {kernel_deps_name}.end()' if needs_synch else ''}));
+  }}
```

Comment:
Use constant size array that we can index into instead of a dynamic vector. We could even generate it as an initializer list:

```cpp
const std::array<cl::Event, 4> events = {
    kernels[0].ExecuteTaskFork(dependencies, dependencies + 3),
    kernels[1].ExecuteTaskFork(dependencies, dependencies + 3),
    kernels[2].ExecuteTaskFork(dependencies, dependencies + 3),
    kernels[3].ExecuteTaskFork(dependencies, dependencies + 3)
}
```